ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -50,22 +74,28 @@ export default class PermissionConnect extends Component {
   }
 
   redirectFlow (accepted) {
-    const { requestAccountTabs } = this.props
+    const { requestAccountTabs, history } = this.props
     const { originName } = this.state
+
     this.setState({
       page: null,
       permissionAccepted: accepted,
     })
 
-    setTimeout(() => {
-      global.platform.currentTab()
-        .then(({ id: currentTabId }) => {
-          global.platform.switchToTab(requestAccountTabs[originName])
-            .then(() => {
-              global.platform.closeTab(currentTabId)
-            })
-        })
-    }, 2000)
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_FULLSCREEN) {
+
+      setTimeout(() => {
+        global.platform.currentTab()
+          .then(({ id: currentTabId }) => {
+            global.platform.switchToTab(requestAccountTabs[originName])
+              .then(() => {
+                global.platform.closeTab(currentTabId)
+              })
+          })
+      }, 2000)
+    } else {
+      history.push(CONNECTED_ROUTE)
```

Comment:
Nite: Maybe it would be better to redirect to home for the notification view? From home, it'll then be redirected back to whatever request may have opened the notification window in the first place (e.g. a transaction). Redirecting to the connected sites page might be confusing and unhelpful if they're expecting a transaction notification. That could happen if they clear permissions or update the extension while the window is open and a transaction is pending, then proceed to re-connect to continue.

I like this idea for the popup view though. It seems helpful and informative, and maybe acts as a good substitute for better communication about when things have succeeded or failed. The badge on the MetaMask icon right above the popup should nudge the user into clicking `x` if they want to see whatever is pending.

Reply:
1b68fb882